src/features/programSample/ProgramSampleModal.tsx

```diff
@@ -234,13 +234,15 @@ const ProgramSample = ({
             logger.info(
                 `${json.step}/${json.amountOfSteps}: ${json.progressPercentage}% - ${json.message}`
             );
-            const amountOfProgress =
-                // Future api of shared will remove undefined steps
-                ((json.step! - 1) / json.amountOfSteps!) * 100 +
-                (1 / json.amountOfSteps!) * json.progressPercentage;
-
-            progress.set(fw as Firmware, amountOfProgress);
-            setProgress(new Map(progress.entries()));
+            if (json.step && json.amountOfSteps) {
+                const amountOfProgress =
```

Comment:
I think shared/nrfutil programming is ensuring these values are defined now, and also calculates the totalProgress for us